botorch/models/utils.py

```diff
@@ -254,8 +254,12 @@ def mod_batch_shape(module: Module, names: List[str], b: int) -> None:
 def gpt_posterior_settings():
     r"""Context manager for settings used for computing model posteriors."""
     with ExitStack() as es:
-        es.enter_context(gpt_settings.debug(False))
-        es.enter_context(gpt_settings.fast_pred_var())
+        gp_debug = gpt_settings.debug.is_default()
+        if gp_debug:
+            es.enter_context(gpt_settings.debug(False))
+        gp_fast_pred_var = gpt_settings.fast_pred_var.is_default()
+        if gp_fast_pred_var:
+            es.enter_context(gpt_settings.fast_pred_var())
```

Comment:
We don't recycle this anywhere, do we? No need to assign the variable then.
```suggestion
        if gpt_settings.debug.is_default():
            es.enter_context(gpt_settings.debug(False))
        if gpt_settings.fast_pred_var.is_default():
            es.enter_context(gpt_settings.fast_pred_var())
```